lib/rspec/rails/matchers/active_job.rb

```diff
@@ -163,7 +163,27 @@ def at_match?(job)
             return job[:at].nil? if @at == :no_wait
             return false unless job[:at]
 
-            values_match?(@at, Time.at(job[:at]))
+            scheduled_at = Time.at(job[:at])
+            values_match?(@at, scheduled_at) || check_for_inprecise_value(scheduled_at)
+          end
+
+          def check_for_inprecise_value(scheduled_at)
+            return unless Time === @at && values_match?(@at.change(usec: 0), scheduled_at)
+
+            RSpec.warn_with((<<-WARNING).gsub(/^\s+\|/, '').chomp)
+            |[WARNING] Your expected `at(...)` value does not match the job scheduled_at value
+            |unless microseconds are removed. This precision error often occurs when checking
+            |values against `Time.current` / `Time.now` which have usec precision, but Rails
+            |uses `n.seconds.from_now` internally which has a usec count of `0`.
+            |
+            |Use `change(usec: 0)` to correct these values.
```

Comment:
Maybe add something that can be copy pasted? Like
```suggestion
            |Use `Time.current.change(usec: 0)` or `Time.now.change(usec: 0)` to correct these values.
```

Reply:
I thought about that, but you can copy paste the addition to your code, as its not necessarily going to be either of those values...

Reply:
Yes completely. 
Maybe:
```suggestion
            |Use `change(usec: 0)` to correct these values. For example with current time: `Time.current.change(usec: 0)`
```

This is not a blocker for me. 😄